src/mbm.jl

```diff
@@ -0,0 +1,405 @@
+################################################################################
+#               CONSTRAINT, DISJUNCTION, DISJUNCT REFORMULATION
+################################################################################
+#Reformulates the disjunction using multiple big-M values
+function reformulate_disjunction(
+    model::JuMP.AbstractModel, 
+    disj::Disjunction,
+    method::MBM
+)
+    ref_cons = Vector{JuMP.AbstractConstraint}() 
+    for d in disj.indicators
+        # Make a copy to avoid modifying during iteration
+        constraints = copy(_indicator_to_constraints(model)[d])
+        for cref in constraints
+            con = JuMP.constraint_object(cref)
+            if con isa JuMP.ScalarConstraint{<:Any, <:MOI.Interval}
+                # Create lower and upper bound constraints
+                lower_con = JuMP.build_constraint(error, con.func, 
+                    MOI.GreaterThan(con.set.lower))
+                upper_con = JuMP.build_constraint(error, con.func, 
+                    MOI.LessThan(con.set.upper))
+                # Create new disjunct constraints
+                JuMP.add_constraint(model,
+                    _DisjunctConstraint(lower_con, d))
+                JuMP.add_constraint(model,
+                    _DisjunctConstraint(upper_con, d))
+                JuMP.delete(model, cref)
+            end 
+            end
+        end
+    for d in disj.indicators
+        _reformulate_disjunct(model, ref_cons, d, 
+            LogicalVariableRef[x for x in disj.indicators if x != d], method)
+    end
+    return ref_cons
+end
+#Reformualates a disjunct the disjunct of interest 
+#represented by lvref and the other indicators in conlvref
+function _reformulate_disjunct(
+    model::JuMP.AbstractModel, 
+    ref_cons::Vector{JuMP.AbstractConstraint}, 
+    lvref::LogicalVariableRef,
+    conlvref::Vector{LogicalVariableRef},
+    method::AbstractReformulationMethod
+)
+    
+    bconref = Dict(d => binary_variable(d) for d in conlvref)
+    !haskey(_indicator_to_constraints(model), lvref) && return
+    constraints = _indicator_to_constraints(model)[lvref]
+    M = Dict{LogicalVariableRef,Float64}()
```

Comment:
Instead of `Float64`, we should use the `value_type`. Please correct all the instances of this.